src/Analyser/ResultCache/ResultCacheManager.php

```diff
@@ -605,6 +603,27 @@ private function save(
 		);
 	}
 
+	/**
+	 * @param mixed[]|null $projectConfig
+	 */
+	private function getCacheFilePath(?string $resultCacheName, ?array $projectConfig): string
+	{
+		if ($resultCacheName !== null) {
+			return $this->tempResultCachePath . '/' . $resultCacheName . '.php';
+		}
+
+		if ($projectConfig !== null &&
+			array_key_exists('parameters', $projectConfig) &&
+			!array_key_exists('resultCachePath', $projectConfig['parameters']) &&
+			(!array_key_exists('tmpDir', $projectConfig['parameters']) || $projectConfig['parameters']['tmpDir'] === null)
```

Comment:
the `tmpDir` case is already covered by an [existing test](https://github.com/phpstan/phpstan-src/blob/d82d7e7fd856360cab395a41282248e4b79ac8a1/tests/e2e/phpstan.neon#L5) which uses a `tmpDir`